### PR TITLE
Support list query for explorer

### PR DESCRIPTION
**Pull request by sooahleex**
### Summary

- Same as title.
- Support list query of DatasetItem and string for explorer

### How to test

- I add some unit tests.

### Checklist

- [X] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: src/datumaro/components/algorithms/hash_key_inference/explorer.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def __init__(
 
     def explore_topk(
         self,
-        query: Union[DatasetItem, str],
+        query: Union[DatasetItem, str, List[DatasetItem], List[str]],
```
Comment (bonhunko) [inline src/datumaro/components/algorithms/hash_key_inference/explorer.py]:
```suggestion
        query: Union[DatasetItem, str, List[Union[DatasetItem, str]]],
```

**Review by bonhunko — APPROVED** [review]
I left a minor comment. LGTM!

File: src/datumaro/components/algorithms/hash_key_inference/explorer.py [hunk 2]
```diff
@@ -74,6 +74,32 @@ def explore_topk(
...
+                    )
+
+            for query_key in query_hash_key_list:
+                unpacked_key = np.unpackbits(query_key.hash_key, axis=-1)
+                logits = calculate_hamming(unpacked_key, database_keys)
+                ind = np.argsort(logits)
+
+                item_list = np.array(self._item_list)[ind]
+                result_list.extend(item_list[:topk_for_query].tolist())
+            return np.random.choice(result_list, topk)
```
Comment (vinnamkim) [inline src/datumaro/components/algorithms/hash_key_inference/explorer.py]: Why is there a non-deterministic process here? Last time you said that the N-to-M top-k similarity search 1) finds the k similarity items for N queries (Nk candidates) and 2) pick up the top-k items from them, is it right?

Comment (sooahleex) [reply src/datumaro/components/algorithms/hash_key_inference/explorer.py]: I updated the process for this. First make N*topk or N*topk_for_query candidates and logits for N queries and sort logits by value. Resort candidates through logit indices and pick up top-k items from them.

**Review by vinnamkim — APPROVED** [review]
LGTM.
